### ISSUE TITLE
[personality-insights] Call to profile() failing in Google stack

### ISSUE THREAD

**Issue by jeffpk62** [watson-developer-cloud/java-sdk#307]
The following simple Personality Insights Java program is failing with a JavaParseException in the Google stack.  The source file is here (renamed for attachment): [PersonalityInsightsExample.txt](https://github.com/watson-developer-cloud/java-sdk/files/291896/PersonalityInsightsExample.txt).  The input file with the profile data is here (renamed from .json for attachment): [p.txt](https://github.com/watson-developer-cloud/java-sdk/files/291865/p.txt).  And the output of the stack trace is here: [out.txt](https://github.com/watson-developer-cloud/java-sdk/files/291866/out.txt). I'm not sure what the issue is with this one; the JSON file appears to conform to the expected input for parsing an argument to the service (the same file works from the command line with cURL), and the code matches what I see in examples.

Testing with **java-sdk-2.10.0-jar-with-dependencies.jar**.

**Comment by germanattanasio** [comment]
You are telling `Gson` that the file has an array of `ContentItem`s but it actually has a `contentItems` field which value is a list of `ContentItem`.

The code you have is:
```java
JsonReader jReader = new JsonReader(new FileReader(file));
List<ContentItem> contentItems = new Gson().fromJson(jReader, TYPE);
ProfileOptions options = new ProfileOptions().contentItems(contentItems);
Profile profile = service.getProfile(options);
System.out.println(profile);
```
That will work if `file` was an array of content items but `file` is an object with a property(`contentItems`) that has the array.

`file:` is:

```json
{
"contentItems": []
}
```
**not** 
```json
[]
```

-----

In order to fix it, you need to parse the object as `Content.java`  which is a class that a `ContentItems` array.

```java
JsonReader jReader = new JsonReader(new FileReader(file));
Content content = new Gson().fromJson(jReader, Content.java);
ProfileOptions options = new ProfileOptions().contentItems(content.getContentItems());
Profile profile = service.getProfile(options);
System.out.println(profile);
```

or get the `contentItems` from the `JsonObject` and then deserializing them using `Gson`

```java
JsonParser parser = new JsonParser();
JsonElement jsonElement = parser.parse(new FileReader(file));
JsonObject jsonObject = jsonElement.getAsJsonObject();
JsonArray jsonArray = jsonObject.getAsJsonArray("contentItems");
List<ContentItem> contentItems = GsonSingleton.getGson().fromJson(jsonArray, TYPE);
ProfileOptions options = new ProfileOptions().contentItems(contentItems);
Profile profile = service.getProfile(options);
System.out.println(profile);
```

**Comment by jeffpk62** [comment]
German, thanks a lot for taking the time to reply and to correct my Json parsing!  I'm reopening with a few follow-up issues.

**1**. The first code snippet you include doesn't compile for me.  I don't know what Content.java is, and apparently, neither does the compiler. :-) 

```
com\ibm\watson\developer_cloud\personality_insights\v2\PersonalityInsightsExample.java:52: error: cannot find symbol
        Content content = new Gson().fromJson(jReader, Content.java);
                                                              ^
  symbol:   variable java
  location: class Content
```

I played around some with the code, but I can't find a definition for Content.java, and I'm just not sure what it is.

**2.** The second code snippet builds and runs but has some problems.

**2.A**  First, it reports the following ParseExceptions on the dates:  [java-errors.txt](https://github.com/watson-developer-cloud/java-sdk/files/302772/java-errors.txt).

**2.B**  Second, the output is different in a few ways from what I get from the API when I test with the same file via cURL.  The cURL output is [curl.txt](https://github.com/watson-developer-cloud/java-sdk/files/302776/curl.txt); the Java output is [java.txt](https://github.com/watson-developer-cloud/java-sdk/files/302777/java.txt).

The first difference is that the Java output does not include any date/time information, which is obviously because of the parse errors.  The second difference is admittedly cosmetic, but the ordering of the fields is markedly different.  I know that this makes absolutely no difference programmatically, which is the primary use of the output.  But it's a little problematic for documenting things, and if there were any way to produce output that's at least closer to the default service API, that would be helpful.  Again, trivial, but I wanted to mention it.

Thanks again, German!

**Comment by germanattanasio** [comment]
**1.** [`Content.java`](https://github.com/watson-developer-cloud/java-sdk/blob/70b51ef3e641419aae824e6976f0e0321c3bf369/src/main/java/com/ibm/watson/developer_cloud/personality_insights/v2/model/Content.java) is part of the `model` package for personality insights.

 **2.A**: @max-vogler can we add the DateSerializer from Tradeoff Analytics to the `ContentItem` class too? That will fix this issue.

 **2.B**:  We can't do anything about this @jeffpk62, I don't know how the service is doing the JSON, and if I want to mimic it I will have to write a custom serializer that will be obsolete everytime they add/remove a field from the service output.

**Comment by max-vogler** [comment]
> 2.A: can we add the DateSerializer from Tradeoff Analytics to the ContentItem class too? That will fix this issue.

Yes, simply add `@JsonAdapter(LongToDateTypeAdapter.class)` to all Date fields (see [CustomVoiceModel](https://github.com/watson-developer-cloud/java-sdk/blob/master/src/main/java/com/ibm/watson/developer_cloud/text_to_speech/v1/model/CustomVoiceModel.java)).

**Comment by germanattanasio** [comment]
done

**Comment by germanattanasio** [comment]
@jeffpk62 I added a test case and fixed the gson deserializing issue.

Take a look at the example below:

```java
JsonReader jReader = new JsonReader(new FileReader("contentitems.json"));
Content content = GsonSingleton.getGson().fromJson(jReader, Content.class);
ProfileOptions options = new ProfileOptions.Builder().contentItems(content.getContentItems()).build();
Profile profile = service.getProfile(options).execute();
```

**Comment by jeffpk62** [comment]
**1.** I'll look again, German; I could easily have missed it.  Is it available with v2.10 or is it new for v3.0?

**2.A** Thanks for jumping on this one, @max-vogler and @germanattanasio .  Will it be fixed in v2.10 or only in v3.0?

**2.B** I understand entirely, German; thanks for the response.  Just thought I'd ask.

**Comment by germanattanasio** [comment]
`1.` it's available in `2.10.0` @jeffpk62

`2.A` v3.0
